x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/data/DocVector.java

```diff
@@ -210,8 +210,15 @@ public static long ramBytesEstimated(
         int[] shardSegmentDocMapForwards,
         int[] shardSegmentDocMapBackwards
     ) {
-        return BASE_RAM_BYTES_USED + RamUsageEstimator.sizeOf(shards) + RamUsageEstimator.sizeOf(segments) + RamUsageEstimator.sizeOf(docs)
-            + RamUsageEstimator.shallowSizeOf(shardSegmentDocMapForwards) + RamUsageEstimator.shallowSizeOf(shardSegmentDocMapBackwards);
```

Comment:
There was a bug in here that I wanted to fix as well. Calling `ramBytesEstimated` after filtering a doc vector created NPEs.